<launch>
  <arg name="pub_mode" default="false"/>
  <arg name="controller" default="true"/>

  <param name="robot_group" type="string" value="ns2"/>
  <group ns ="ns2">
    <param name="issimulation" type="bool" value="true"/>
    
    <node name="mujoco_ros" pkg="mujoco_ros" type="mujoco_ros" required="true" respawn="false" output="screen">
        <param name="license" type="string" value="$(env HOME)/.mujoco/mjkey.txt" />
        <param name="pub_mode" value="$(arg pub_mode)"/>
        <param name="model_file" type="string" value="$(find scout_panda_description)/robots/scout_panda_ns2.xml"/>
    </node>
        <node name="kimm_action_manager_simulation" pkg="kimm_action_manager" args="-keyboard" type="kimm_action_manager_simulation" output="screen">
    </node>

    <param name="urdf_path" type="string" value="$(find scout_panda_description)"/>
    <param name="urdf_name" type="string" value="/robots/scout_panda_hand_free.urdf"/>
    <param name="gui_urdf_path" type="string" value="$(find scout_panda_description)"/>
    <param name="gui_urdf_name" type="string" value="/robots/scout_panda_rviz_ns2.urdf"/>
    <arg name="description_name"           default="robot_description"/>
    <arg name="description_file"           default="$(find scout_panda_description)/robots/scout_panda_rviz_ns2.urdf"/>
    <param name="$(arg description_name)" textfile="$(arg description_file)"/>

    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="odom" args="0 0 0 0 0 0 odom ns0_rviz_base_link" /> -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true">
			<param name="publish_frequency" value="100" />
			<param name="rate" value="100" />
		</node>
    <include file="$(find kimm_husky_navigation)/launch/move_base_mapless_demo_ns2.launch"/>
  </group>

</launch>
